### PR TITLE
Replace `_computedTimeWindowPart` with `_time`

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

Use absolute `_time` instead of `_computedTimeWindowPart` for time tracking.

## Motivation and additional information

Improves readability.

## Author's checklist

* [x] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] I sticked to C++17 features.
* [x] I sticked to CMake version 3.16.3.
* [x] I squashed / am about to squash all commits that should be seen as one.

**not relevant, because of refactoring**

* [ ] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [ ] I added a test to cover the proposed changes in our test suite.
* [ ] For breaking changes: I documented the changes in the appropriate [porting guide](https://precice.org/couple-your-code-porting-overview.html).

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?

### CODE REVIEW

File: src/cplscheme/BaseCouplingScheme.cpp [hunk 1]
```diff
@@ -766,7 +758,7 @@ void BaseCouplingScheme::advanceTXTWriters()
 
 bool BaseCouplingScheme::reachedEndOfTimeWindow() const
 {
-  return math::equals(_timeWindowSize - _computedTimeWindowPart, 0.0) || not hasTimeWindowSize();
+  return math::equals(_timeWindowStartTime + _timeWindowSize - _time, 0.0) || not hasTimeWindowSize();
```
Comment (fsimonis) [inline src/cplscheme/BaseCouplingScheme.cpp:761]: Why not?
```suggestion
  return math::equals(_timeWindowStartTime + _timeWindowSize, _time) || not hasTimeWindowSize();
```

File: src/cplscheme/BaseCouplingScheme.cpp [hunk 2]
```diff
@@ -348,7 +345,7 @@ void BaseCouplingScheme::secondExchange()
         // The computed time window part equals the time window size, since the
         // time window remainder is zero. Subtract the time window size and do another
         // coupling iteration.
-        PRECICE_ASSERT(math::greater(_computedTimeWindowPart, 0.0));
+        PRECICE_ASSERT(math::greater(_time - _timeWindowStartTime, 0.0));
```
Comment (fsimonis) [inline src/cplscheme/BaseCouplingScheme.cpp:348]: Why not?
```suggestion
        PRECICE_ASSERT(math::greater(_time, _timeWindowStartTime));
```

File: src/cplscheme/BaseCouplingScheme.cpp [hunk 3]
```diff
@@ -197,32 +197,29 @@ void BaseCouplingScheme::receiveData(const m2n::PtrM2N &m2n, const DataMap &rece
...
+  // buffer current time
+  const double oldTime = _time;
+  // set _time state to point to end of this window such that _time in receiveData is at end of window
+  _time += _nextTimeWindowSize;
   // receive data for end of window
   this->receiveData(m2n, receiveData);
   // reset time state;
-  _computedTimeWindowPart = oldComputedTimeWindowPart;
-  _timeWindowStartTime    = oldTimeWindowStartTime;
+  _time = oldTime;
```
Comment (fsimonis) [inline src/cplscheme/BaseCouplingScheme.cpp:214]: Is there a reason we still need this hack?

Comment (fsimonis) [reply src/cplscheme/BaseCouplingScheme.cpp:214]: Ugly ugly but at least the hack is obvious now.

**Review by fsimonis — APPROVED** [review]

**Review by uekerman — APPROVED** [review]
Does indeed improve readability :+1:

**Pull request merged**